src/evict/evict_lru.c

```diff
@@ -470,33 +470,34 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
 
     __wt_epoch(session, &now);
 
-#define WT_CACHE_STUCK_TIMEOUT_MS (300 * WT_THOUSAND)
-    time_diff_ms = WT_TIMEDIFF_MS(now, cache->stuck_time);
-
+    /* The checks below should only be executed when a cache timeout has been set. */
+    if (cache->cache_stuck_timeout_ms > 0) {
+        time_diff_ms = WT_TIMEDIFF_MS(now, cache->stuck_time);
 #ifdef HAVE_DIAGNOSTIC
-    /* Enable extra logs 20ms before timing out. */
-    if (time_diff_ms > WT_CACHE_STUCK_TIMEOUT_MS - 20) {
-        WT_SET_VERBOSE_LEVEL(session, WT_VERB_EVICT, WT_VERBOSE_DEBUG_1);
-        WT_SET_VERBOSE_LEVEL(session, WT_VERB_EVICTSERVER, WT_VERBOSE_DEBUG_1);
-        WT_SET_VERBOSE_LEVEL(session, WT_VERB_EVICT_STUCK, WT_VERBOSE_DEBUG_1);
-    }
+        /* Enable extra logs 20ms before timing out. */
+        if (time_diff_ms > cache->cache_stuck_timeout_ms - 20) {
```

Comment:
Seeing subtraction like this makes me nervous about underflow. Could you update the code to ensure that's not possible (i.e: that `cache_stuck_timeout_ms` is always greater than 20)?

You could manage that either via making the `20` a define somewhere and checking initial values against in on setup, or using a `(time_diff_ms > cache->cache_stuck_timeout_ms - (WT_MIN(cache->cache_stuck_timeout_ms, 20))` clause instead.

Reply:
I agree that this is necessary especially if we decide to let the user set the timeout in ms instead of seconds.

Reply:
As discussed, we will always enable extra logs when the timeout is less than 20ms, see here: 2660ae0467d2a986cdcd1601304651ebf5d255c9.